gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/upgrader/PlanApiTypeUpgrader.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.service.impl.upgrade.upgrader;
+
+import io.gravitee.definition.model.DefinitionVersion;
+import io.gravitee.definition.model.v4.ApiType;
+import io.gravitee.node.api.upgrader.Upgrader;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.ApiRepository;
+import io.gravitee.repository.management.api.PlanRepository;
+import io.gravitee.repository.management.api.search.ApiCriteria;
+import io.gravitee.repository.management.api.search.ApiFieldFilter;
+import io.gravitee.repository.management.model.Api;
+import io.gravitee.repository.management.model.Plan;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Sergii ILLICHEVSKYI (sergii.illichevskyi at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@Component
+public class PlanApiTypeUpgrader implements Upgrader {
+
+    @Lazy
+    @Autowired
+    private ApiRepository apiRepository;
+
+    @Lazy
+    @Autowired
+    private PlanRepository planRepository;
+
+    @Override
+    public boolean upgrade() {
+        try {
+            updatePlanApiType();
+        } catch (Exception e) {
+            log.error("Unable to apply upgrader {}", getClass().getSimpleName(), e);
+            return false;
+        }
+
+        return true;
+    }
+
+    private void updatePlanApiType() throws TechnicalException {
+        log.info("Starting migration of plan api_type...");
+
+        Collection<Api> v4Apis = apiRepository
+            .search(
+                new ApiCriteria.Builder().definitionVersion(List.of(DefinitionVersion.V4)).build(),
+                null,
+                ApiFieldFilter.defaultFields()
+            )
+            .collect(Collectors.toSet());
```

Comment:
Hum for performance reasons.
I think it would be better to use the stream and avoid the `for` on the apis. 
I think this would free up memory during the treated apis

🤔  and it would be a good idea to test the upgrade on master data first, to make sure it works well on many api and plans.

Reply:
Done